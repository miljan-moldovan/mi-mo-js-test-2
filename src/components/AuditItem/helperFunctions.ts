import moment from 'moment';

const AuditType = {
  NULL: -1,
  Normal: 0,
  Change: 1,
  Cancel: 2,
  ReviewWeb: 10,
};

const DateTime = {
  time: 'HH:mm:ss',
  timeOld: 'HH:mm',
  displayTime: 'hh:mm A',
  date: 'YYYY-MM-DD',
  dateTime: 'YYYY-MM-DDT00:00:00',
  serverDateTime: 'YYYY-MM-DDTHH:mm:ss',
};

export const getAuditType = (auditType?: number):string => {
  switch (auditType) {
    case AuditType.Normal:
      return 'BOOKED';
    case AuditType.Change:
      return 'MODIFIED';
    case AuditType.Cancel:
      return 'CANCEL';
    case AuditType.ReviewWeb:
      return 'REVIEW-WEB';
    default:
      return '--';
  }
};

export const formatDate = (date: string):string =>
  moment(date, DateTime.date).format('MM/DD/YYYY');

export const formatTime = (time: string):string =>
  moment(time, DateTime.time).format(DateTime.displayTime);

export const formatTimeWithMinutes = (time: string):string =>
  moment(time, DateTime.serverDateTime).format(DateTime.displayTime);

export const formatEmployeeName = (employee?: string):string => {
  if (!employee) {
    return '';
  }
  const splitName = employee.split(' ');
  switch (splitName.length) {
    case 1:
      return splitName[0];
    case 2:
      return `${splitName[0]} ${splitName[1][0]}`;
    default:
      return `${splitName[0]} ${splitName[2][0]}`;
  }
};
