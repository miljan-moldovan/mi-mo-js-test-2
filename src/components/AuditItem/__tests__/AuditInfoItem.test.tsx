import * as React from 'react';
import { mount } from 'enzyme';
import AuditInfoItem from '../index';
import { formatDate, formatTime, formatEmployeeName } from '../helperFunctions';

const auditMock = {
  appointmentDate: '2018-12-19T00:00:00',
  appointmentEndTime: '15:15:00',
  appointmentStartTime: '14:45:00',
  auditDateTime: '2018-12-19T14:53:51',
  auditEmployee: 'Beth',
  auditType: 1,
  id: 1719095,
  provider: 'Beth',
  service: "Men's Cut",
};

describe('<AuditInfoItem />', () => {
  let defaultProps;

  // just for remove error with depend using react-dom for render. Need it before use adapter for RN
  const origConsole = console.error;
  const mountComponent = (props = defaultProps) => {
    return mount(<AuditInfoItem {...props} />);
  };

  beforeEach(() => {
    // just for remove error with depend using react-dom for render. Need it before use adapter for RN
    console.error = () => {};
    defaultProps = {
      isBlockTime: false,
      singleAudit: auditMock,
    };
  });
  afterEach(() => {
    // just for remove error with depend using react-dom for render. Need it before use adapter for RN
    console.error = origConsole;
  });

  it('renders correctly', () => {
    expect(mountComponent()).toMatchSnapshot();
  });

  describe('Should render text in different formats depending on `isBlockTime`', () => {
    it('`isBlockTime` is `true`', () => {
      defaultProps.isBlockTime = true;
      const renderedName = mountComponent()
        .find('Text')
        .last()
        .text();
      expect(renderedName).toBe(formatEmployeeName(auditMock.auditEmployee));
    });
    it('`isBlockTime` is `false`', () => {
      defaultProps.isBlockTime = false;
      auditMock.service = null;

      const renderedString = `${''} with ${
        formatEmployeeName(auditMock.provider)} on ${
          formatDate(auditMock.appointmentDate)} at ${' '}${
            formatTime(auditMock.appointmentStartTime)}`;

      const renderedName = mountComponent()
        .find('Text')
        .someWhere(n => n.text() === renderedString);

      expect(renderedName).toBe(true);
    });
  });
  describe('Should render auditInfoText in different formats depending on `isBlockTime`', () => {
    it('`isBlockTime` is `true`', () => {
      defaultProps.isBlockTime = true;
      const renderedName = mountComponent()
        .find('Text')
        .last()
        .text();
      expect(renderedName).toBe(formatEmployeeName(auditMock.auditEmployee));
    });
    it('`isBlockTime` is `false`', () => {
      defaultProps.isBlockTime = false;
      const renderedName = mountComponent()
        .find('Text')
        .last()
        .text();
      expect(renderedName).not.toBe(formatEmployeeName(auditMock.auditEmployee));
    });
  });
});
