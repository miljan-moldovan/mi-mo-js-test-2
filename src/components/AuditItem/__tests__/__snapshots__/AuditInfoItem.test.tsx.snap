// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuditInfoItem /> renders correctly 1`] = `
<AuditInfoItem
  isBlockTime={false}
  singleAudit={
    Object {
      "appointmentDate": "2018-12-19T00:00:00",
      "appointmentEndTime": "15:15:00",
      "appointmentStartTime": "14:45:00",
      "auditDateTime": "2018-12-19T14:53:51",
      "auditEmployee": "Beth",
      "auditType": 1,
      "id": 1719095,
      "provider": "Beth",
      "service": "Men's Cut",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingVertical": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingVertical": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 14,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": 14,
            }
          }
        >
          MODIFIED
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#3F3F3F",
            "fontFamily": "Roboto",
            "fontSize": 11,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#3F3F3F",
              "fontFamily": "Roboto",
              "fontSize": 11,
            }
          }
        >
          Men's Cut
           with 
          Beth
           on 
          12/19/2018
           at 
           
          02:45 PM
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "Roboto",
            "fontSize": 11,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": 11,
            }
          }
        >
          by 
          Beth
           on 
          12/19/2018
        </Text>
      </Text>
    </View>
  </View>
</AuditInfoItem>
`;
